File: derekbanas/js/02_bubble_sort.js
function swap(inputs, i, j) {
    var tmp = inputs[i];
    inputs[i] = inputs[j];
    inputs[j] = tmp;
}


function bubbleSort( inputs ) {
    for ( var i = inputs.length - 1; i >= 0; i --) {
        for ( var j = 0; j < i; j ++) {
            if (inputs[j] > inputs[j + 1]) {
                swap(inputs, j, j + 1);
            }
        }
    }
}




var inputs = [
    1, 23, 4, 434, 232, 324, 11, 2323, 5
];

bubbleSort(inputs);

console.log(inputs);

File: derekbanas/python/02_bubble_sort.py


def bubble_sort(inputs):

    for i in range(len(inputs)-1, -1, -1):
        for j in range(0, i):
            if inputs[j] > inputs[j + 1] :
                swap(inputs, j, j + 1)





def swap(inputs, i, j) :
    tmp = inputs[i]
    inputs[i] = inputs[j]
    inputs[j] = tmp



if __name__ == '__main__':
    inputs = [
        1, 23, 4, 434, 232, 324, 11, 2323, 5
    ]
    bubble_sort(inputs)
    print inputs